### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by kelson42**

### CODE REVIEW

**Comment by kelson42** [comment]
@rgaudin I believe to have configured the `workflow_dispatch` properly... but somehow it does not appear on the UI and I was not able to test it properly.

**Comment by rgaudin** [comment]
> @rgaudin I believe to have configured the `workflow_dispatch` properly... but somehow it does not appear on the UI and I was not able to test it properly.

Needs to be on main branch

**Comment by kelson42** [comment]
@rgaudin Can I merge then?

**Review by rgaudin — CHANGES_REQUESTED** [review]
See inline comments

File: .github/workflows/build-and-upload.yml [hunk 1]
```diff
@@ -7,6 +7,12 @@ on:
     branches:
       - main
   pull_request:
+  workflow_dispatch:
+    inputs:
+      publish-to-s3:
+        description: Publish base Image to S3, for testing purpose
+        required: false
+        default: 1
```
[the remark this answers is not in the document]
Comment (kelson42) [reply .github/workflows/build-and-upload.yml:15]: fixed in latest commit

File: .github/workflows/build-and-upload.yml [hunk 2]
```diff
@@ -45,7 +51,7 @@ jobs:
           ./builder.py --arch "${{ matrix.arch }}" --compress --output "${{ env.BASE_IMAGE_FILENAME_RADICAL }}"
 
       - name: Upload base image to S3
-        if: startsWith(github.ref, 'refs/tags/') != true
+        if: ${{ github.event.inputs.publish-to-s3 }}
```
[the remark this answers is not in the document]
Comment (kelson42) [reply .github/workflows/build-and-upload.yml:54]: fixed in latest commit

**Review by rgaudin — APPROVED** [review]
LGTM

**Pull request merged**